pkg/acsengine/types.go

```diff
@@ -24,3 +24,36 @@ type V20160330ContainerService struct {
 	api.TypeMeta
 	*v20160330.ContainerService
 }
+
+//DCOSSpecConfig is the configurations of DCOS
+type DCOSSpecConfig struct {
+	DCOS173_BootstrapDownloadURL string
+	DCOS184_BootstrapDownloadURL string
+	DCOS187_BootstrapDownloadURL string
+}
+
+//DockerSpecConfig is default docker install script URL.
+type DockerSpecConfig struct {
+	DefaultDockerInstallScriptURL string
+}
+
+//KubernetesSpecConfig is the kubernetes container images used.
+type KubernetesSpecConfig struct {
+	DefaultKubernetesHyperkubeSpec         string
```

Comment:
Thanks for incorporating feedback comments. I still see KubernetesSpecConfig with 10 properties. It seems to me that for every new cloud these 10 fields will need to be added and someone can selectively override some of these fields. I am not sure if we want to give this much configurability to the caller. 

Also, this could become unmanageable/will need to versioned if a new spec config needs to be introduced.

In my earlier feedback comment what I meant was: can we expect the caller to give us the name of the base container like: "gcr.io/google_containers" or "mirror.azure.cn:5000/google_containers". And we also expect the name of spec to be a const.

Then for e.g. kubernetesDNSMasqSpec can be constructed by concatenating the base URL (provided by the caller): "mirror.azure.cn:5000/google_containers" and a constant string: "/kube-dnsmasq-amd64:1.3"  

Let me know if this makes sense or I missed something.